### ISSUE TITLE
[Feature] Update reason-react to jsx 3

### ISSUE THREAD

**Issue by arecvlohe** [phenomic/phenomic#1278]
Hi @MoOx! I love using this library so thank you very much for creating it.

I am upgrading my phenomic site to use hooks in ReasonReact. There are a couple things that need updating I believe. I narrowed down what I think needs updating to this file:

```reason
[@bs.module "@phenomic/plugin-renderer-react/lib/client"]
external createContainer_ :
  (ReasonReact.reactClass, Js.t({..})) => ReasonReact.reactClass =
  "createContainer";

[@bs.module "@phenomic/plugin-renderer-react/lib/client"]
external withPhenomicApi_ :
  (ReasonReact.reactClass, Js.t({..})) => ReasonReact.reactClass =
  "withPhenomicApi";

[@bs.module "@phenomic/plugin-renderer-react/lib/client"]
external withInitialProps_ :
  (ReasonReact.reactClass, Js.t({..})) => ReasonReact.reactClass =
  "withInitialProps";

module BodyRenderer = BodyRenderer;

module Link = Link;

let createContainer = (comp, queries) => createContainer_(comp, queries);

let withPhenomicApi = (comp, queries) => withPhenomicApi_(comp, queries);

let withInitialProps = comp => withInitialProps_(comp);

type jsNodeList('a) = {
  .
  "list": array('a),
  "previousPageIsFirst": bool,
  "previous": Js.nullable(string),
  "next": Js.nullable(string),
};

type edge('a) =
  | Idle('a)
  | Loading
  | Inactive
  | Errored;

type jsEdge('a) = {
  .
  "status": string,
  "node": 'a,
};

let jsEdge = jsEdge =>
  switch (jsEdge##status) {
  | "loading" => Loading
  | "error" => Errored
  | "idle" => Idle(jsEdge##node)
  | "inactive"
  | _ => Inactive
  };

```

From what I can tell, `ReasonReact.reactClass = "withPhenomicApi"` would become `React.element = "withPhenomicApi"` and I would need to update `Link.re` and `BodyRenderer.re` as well. With those updates it should work as expected?

Anyhow, I would like to help submit a PR for this issue but first just wondering if I am on the right path.

**Comment by MoOx** [comment]
@arecvlohe Indeed we should totally update all reason code to be jsx3 ready. I don't have time atm so if you are up for a pr that would be awesome!

**Comment by arecvlohe** [comment]
@MoOx Cool! I will work on this a bit and see where I can get it. I will let you know if I run into any issues.

**Comment by MoOx** [comment]
@arecvlohe do not hesitate to reach me if you have any question or need help.

**Comment by MoOx** [comment]
@arecvlohe did you started something? I would like to work on that to get it asap. So if you haven't I can start, otherwise, if you started, you could open a PR so we can both work on it :)

**Comment by arecvlohe** [comment]
@MoOx I started to look at this last night and got your email this evening so I opened up this PR: https://github.com/phenomic/phenomic/pull/1279.

The dependency chain, or how you install `reason-react` globally without setting it in `package.json`, has confused me so I have spent more time trying to figure that out than trying to work on the code itself. I have submitted a few changes that I think are correct but it still is in a bad state. Maybe if you look at my changes you can help guide me on what I need to do to handle dependencies in a way that is expected in the repo.

**Comment by arecvlohe** [comment]
Implemented in #1279
